core/src/test/scala/integration/kafka/api/SaslSetup.scala

```diff
@@ -81,4 +84,14 @@ trait SaslSetup {
     System.clearProperty("zookeeper.authProvider.1")
     Configuration.setConfiguration(null)
   }
+
+  def kafkaSaslProperties(clientSaslMechanism: String, serverSaslMechanisms: List[String] = List.empty) = {
+    val props = new Properties
+    props.put(SaslConfigs.SASL_MECHANISM, clientSaslMechanism)
+    if (!serverSaslMechanisms.isEmpty) {
+      props.put(KafkaConfig.SaslMechanismInterBrokerProtocolProp, clientSaslMechanism)
+      props.put(SaslConfigs.SASL_ENABLED_MECHANISMS, serverSaslMechanisms.mkString(","))
```

Comment:
Are there cases in tests where we actually want to set this to an empty list? Maybe `saslServerMechanisms` should be `Option[Seq[String]]` and we use `None` to mean what we currently use empty list for.